### PR TITLE
[datadogexporter] Add env variables support for config options

### PR DESCRIPTION
**Pull request by KSerrania**
**Description:** Uses the feature added in https://github.com/open-telemetry/opentelemetry-collector/pull/2231 to allow users to set config options with environment vairables.

Allowed environment variables:
- `DD_API_KEY` -> `Config.API.Key`
- `DD_SITE` -> `Config.API.Site`
- `DD_HOST` -> `Config.TagsConfig.Hostname`
- `DD_ENV` -> `Config.TagsConfig.Env`
- `DD_SERVICE` -> `Config.TagsConfig.Service`
- `DD_VERSION` -> `Config.TagsConfig.Version`
- `DD_TAGS` -> `Config.TagsConfig.EnvVarTags` (replaces `Config.TagsConfig.Tags` if not set)
- `DD_URL` -> `Config.Metrics.TCPAddr.Endpoint`
- `DD_APM_URL` -> `Config.Traces.TCPAddr.Endpoint`

For `DD_TAGS`, since `Config.TagsConfig.Tags` is a map, and the env variable replacement only operates on string fields, a new placeholder string option `Config.TagsConfig.EnvVarTags` was created to temporarily get the environment variable content.

**Link to tracking Issue:**  n/a

**Testing:** Added unit tests, and tested in our exporter E2E test environment

**Documentation:** Updated the example config to explain which environment variables can be used.

### CODE REVIEW

File: exporter/datadogexporter/config/config.go
Comment (mx-psi) [inline exporter/datadogexporter/config/config.go:117]: If we make this field private it's ignored by `mapstructure` and we avoid an user manually setting this field. This means adding some methods to `TagsConfig` but I think it's cleaner.

Comment (KSerrania) [reply exporter/datadogexporter/config/config.go:117]: This wouldn't work because If we make this field private, it's also ignored by the environment variable replacement logic (https://github.com/open-telemetry/opentelemetry-collector/blob/51563860af5a558b6e7087feea2502b5d348375c/config/config.go#L708), because you can't externally modify unexported private fields.

We'd need a custom `(t *TagsConfig)`method to expand this field, which reuses the env var expansion logic somehow, and call it manually at an appropriate time (probably in `CreateDefaultConfig`, but that means the expansion is not done at the same time as for the other env variables). I'm not sure which solution is cleaner here.

Comment (mx-psi) [reply exporter/datadogexporter/config/config.go:117]: > This wouldn't work because If we make this field private, it's also ignored by the environment variable replacement logic (https://github.com/open-telemetry/opentelemetry-collector/blob/51563860af5a558b6e7087feea2502b5d348375c/config/config.go#L708), because you can't externally modify unexported private fields.

Ah, hadn't thought about that. Yeah, then there's not much benefit on doing it differently.

**Review by mx-psi — APPROVED** [review]
LGTM after fixing the merge conflict

**Comment by tigrannajaryan** [comment]
We allow using env variables in the config file, e.g.:

```yaml
 datadog:	  
    hostname: "$DD_HOST"
```

You can ship a default config file that has all env vars specified like this.

Will this serve your needs and make this PR unnecessary?

**Comment by KSerrania** [comment]
Hey @tigrannajaryan,

Indeed, we could use a default file, but in that case, instead of being automatic, the env variable support would work only if users do use the default config file (& do not accidentally remove / alter env variable fields). It works but there’s more room for user error.
Moreover, this means that when we add a new option (& an associated env variable), we have to hope that users will merge their current config file with the new default file when they update their collector. Thus, this is more work for users when they want to upgrade.
Finally, the system proposed in this PR (where env vars are automatically taken into account) is similar to what other Datadog products do, such as the [datadog-agent](https://github.com/DataDog/datadog-agent).

These are the main reasons why we proposed https://github.com/open-telemetry/opentelemetry-collector/pull/2231 (already merged) and this PR.

**Review by tigrannajaryan — APPROVED** [review]
LGTM

**Pull request merged**
